### PR TITLE
Fuzzy search on genes

### LINKED ISSUE #400: Add-a-gene is case sensitive
the "Add" textbox for SELECTED GENES requires correct capitalization of the gene name

Is that the desired behavior?

Ie, entering (on the TM_droplet data):
* `UMOD` - fails
* `Umod` - works

### PR DESCRIPTION
**Pull request by colinmegill**
image

Resolves #400

### CODE REVIEW

File: client/src/components/geneExpression/index.js [hunk 1]
```diff
@@ -4,14 +4,51 @@
...
 import { postUserErrorToast } from "../framework/toasters";
 import ExpressionButtons from "./expressionButtons";
 
+const renderGene = (fuzzySortResult, { handleClick, modifiers, query }) => {
+  if (!modifiers.matchesPredicate) {
+    return null;
+  }
+  /* the fuzzysort wraps the object with other properties, like a score */
+  const gene = fuzzySortResult.obj;
+  // fuzzysort.highlight(fuzzysort.single('tt', 'test'), '*', '*') // *t*es*t*
```
Comment (bkmartinjr) [inline client/src/components/geneExpression/index.js]: pls clean up debug/test code :-)

File: client/src/components/geneExpression/index.js [hunk 2]
```diff
@@ -4,14 +4,51 @@
...
+};
+
+const filterGenes = (query, genes) => {
+  /* fires on load, once, and then for each character typed into the input */
+  return fuzzysort.go(query, genes, {
+    key: "name",
+    limit: 5,
+    threshold: -10000 // don't return bad results
+  });
+  // return `${gene.name.toLowerCase()}`.indexOf(query.toLowerCase()) >= 0;
```
Comment (bkmartinjr) [inline client/src/components/geneExpression/index.js]: cleanup pls

File: client/src/components/geneExpression/index.js [hunk 3]
```diff
@@ -166,3 +187,9 @@ class GeneExpression extends React.Component {
 }
 
 export default GeneExpression;
+
```
Comment (bkmartinjr) [inline client/src/components/geneExpression/index.js]: cleanup

**Comment by colinmegill** [comment]
Update: 

- [ ] works on cord blood dataset

**Pull request merged**